.github/workflows/build.yml

```diff
@@ -0,0 +1,51 @@
+name: Build and test
+
+on:
+  pull_request:
+    branches: [ main ]
+
+jobs:
+  build:
+    strategy:
+      matrix:
+        os: [ubuntu-latest, windows-latest, macos-latest]
+
+    runs-on: ${{ matrix.os }}
+    steps:
+      # PR builds don't checkout the commit by default, so we're going to correct that
```

Comment:
Are you sure this is correct? Testing on my own repositories I don't need to explicitly set the ref when the action is triggered by a PR?

Reply:
yep! taken from the docs for the `pull_request` action:

> Note that GITHUB_SHA for this event is the last merge commit of the pull request merge branch. If you want to get the commit ID for the last commit to the head branch of the pull request, use github.event.pull_request.head.sha instead.

so as a matter of practice now I always start with a checkout

Reply:
I generally like to run my pipelines against the code the user actually pushed, so that there's no surprises from the user's perspective. This works better if only fast-forward rebases are allowed to happen,  IMO.

Reply:
Ah good to know. Thanks!